xmppserver/src/main/java/org/jivesoftware/database/SequenceManager.java

```diff
@@ -150,23 +155,29 @@ public SequenceManager(int seqType, int size) {
         managers.put(seqType, this);
         this.type = seqType;
         this.blockSize = size;
-        currentID = 0L;
-        maxID = 0L;
     }
 
     /**
      * Returns the next available unique ID. Essentially this provides for the functionality of an
      * auto-increment database field.
      * @return the next sequence number
      */
-    public synchronized long nextUniqueID() {
-        if (!(currentID < maxID)) {
-            // Get next block -- make 5 attempts at maximum.
-            getNextBlock(5);
+    public long nextUniqueID() {
+        final Lock lock = sequenceBlocks.getLock(type);
+        lock.lock();
+        try {
+            Data data = sequenceBlocks.get(type);
+            if (data == null || !(data.getCurrentID() < data.getMaxID())) {
```

Comment:
`!(data.getCurrentID() < data.getMaxID())`

Seems like an odd way of expressing this - is there a good reason for it being this way around?

Reply:
I like that it expresses the actual intent. The CurrentID _should_ be less than the MaxID and isn't.
Although by that logic, the line should begin with a check that the data isn't not null 😂

Reply:
@matt-tucker deviced it like that in 2004. Who am I to argue with syntax that survived for 16 years? :)